modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/artifact/metadata/MetadataKeyDeclarationResolver.java

```diff
@@ -68,15 +89,31 @@ public MetadataKeyResult resolveKeyResult() {
     Map<String, String> keyPartValues =
         getMetadataKeyPartsValuesFromComponentDeclaration(componentElementDeclaration, componentModel);
     for (MetadataKeyInfo keyInfo : keyPartModelsInfo) {
-      String id;
+      String id = null;
+      boolean isNullValue = false;
       if (keyPartValues.containsKey(keyInfo.parameterModel.getName())) {
         id = keyPartValues.get(keyInfo.parameterModel.getName());
+        isNullValue = id == null;
+      } else if (!partialKey && keyInfo.parameterModel.getDefaultValue() != null) {
+        String defaultValue = keyInfo.parameterModel.getDefaultValue().toString();
+        if (expressionManager != null && isExpression(defaultValue)) {
```

Comment:
why this validation? When is the expressionManager going to be null? and why shouldn't we fail if that is the case?

Reply:
We use this resolver from MTS to validate that the key is "complete" before calling Runtime. So, on that scenario we won't need to resolve the expression.